core/src/utils/test/hardware-back-button.spec.ts

```diff
@@ -1,11 +1,12 @@
+import type { BackButtonEvent } from '../../../src/interface';
 import { startHardwareBackButton } from '../hardware-back-button';
 
 describe('Hardware Back Button', () => {
   beforeEach(() => startHardwareBackButton());
   it('should call handler', () => {
     const cbSpy = jest.fn();
     document.addEventListener('ionBackButton', (ev) => {
-      ev.detail.register(0, cbSpy);
+      (ev as BackButtonEvent).detail.register(0, cbSpy);
```

Comment:
Then remove it here?

```suggestion
      ev.detail.register(0, cbSpy);
```